onnxruntime/core/providers/xnnpack/xnnpack_execution_provider.cc

```diff
@@ -87,14 +88,38 @@ std::unique_ptr<KernelRegistry> RegisterKernels() {
 using namespace xnnpack;
 
 XnnpackExecutionProvider::XnnpackExecutionProvider(const XnnpackExecutionProviderInfo& info)
-    : IExecutionProvider{kXnnpackExecutionProvider, true} {
+    : IExecutionProvider{kXnnpackExecutionProvider, true}, info_(info) {
   if (info.xnn_thread_pool_size > 1) {
     // pthreadpool is independent of ort-threadpoool, so we have to disable cpu spinning for ort-threadpool.
     // otherwise, the pthreadpool will be starved and harm performance a lot.
     xnnpack_thread_pool_ = pthreadpool_create(static_cast<size_t>(info.xnn_thread_pool_size));
   }
 }
 
+common::Status XnnpackExecutionProvider::CheckSessionOptions(const SessionOptions& so) {
+  if (so.execution_mode != ExecutionMode::ORT_SEQUENTIAL) {
+    return ORT_MAKE_STATUS(ONNXRUNTIME, FAIL,
+                           "Parallel execution mode does not support the Xnnpack Execution Provider. "
+                           "Please set the execution mode as sequential for this "
+                           "session since it uses the Xnnpack Execution Provider.");
+  }
+  if (so.intra_op_param.allow_spinning && so.intra_op_param.thread_pool_size > 1) {
+    LOGS_DEFAULT(WARNING)
+        << "XNNPACK EP utilize pthreadpool for multi-threading. So, if allow_spinning on ORT's"
+           "thread-pool and its pool size is not 1, "
+           "pthreadpool will content with ORT's intra-op thread pool and hurt performance a lot. "
+           "Please Setting intra_op_param.allow_spinning to false or "
+           "setting ort's pool size (intra_thread_num) to 1 and try again.";
+  }
+  // 0 means user didn't set the value.
```

Comment:
nit: leave a space between completely separate blocks so it's easier to skim between them. when they're joined together it takes time to process whether it's a new 'if' or a continuation of the previous one (like an 'else if' or 'if').

Reply:
Done